src/server.rs

```diff
@@ -587,14 +588,13 @@ impl<C: CommandCreatorSync> SccacheServer<C> {
         // Note that we cap the amount of time this can take, however, as we
         // don't want to wait *too* long.
         runtime
-            .block_on(Timeout::new(Compat::new(wait), Duration::new(30, 0)))
-            .map_err(|e| {
-                if e.is_inner() {
-                    e.into_inner().unwrap()
-                } else {
-                    io::Error::new(io::ErrorKind::Other, e)
-                }
-            })?;
+            .block_on_std(async {
+                time::timeout(Duration::new(30, 0), wait)
```

Comment:
does it indicate waiting 30 seconds?

If yes please update info log at line 579